src/main/java/com/paneedah/weaponlib/WeaponSpawnEntity.java

```diff
@@ -81,10 +84,7 @@ protected void onImpact(final RayTraceResult position) {
             //PostProcessPipeline.createDistortionPoint((float) position.hitVec.x,(float)  position.hitVec.y, (float) position.hitVec.z, 2f, 3000);
             Explosion.createServerSideExplosion(weapon.getModContext(), world, this.getThrower(), this, position.hitVec.x, position.hitVec.y, position.hitVec.z, explosionRadius, false, true, isDestroyingBlocks, explosionParticleAgeCoefficient, smokeParticleAgeCoefficient, explosionParticleScaleCoefficient, smokeParticleScaleCoefficient, weapon.getModContext().getRegisteredTexture(explosionParticleTextureId), weapon.getModContext().getRegisteredTexture(smokeParticleTextureId), weapon.getModContext().getExplosionSound());
         } else if (position.entityHit != null) {
-            if (this.getThrower() != null)
-                position.entityHit.attackEntityFrom(DamageSource.causeThrownDamage(this, this.getThrower()), damage);
-            else
-                position.entityHit.attackEntityFrom(new DamageSource("thrown"), damage);
+            position.entityHit.attackEntityFrom(new ProjectileDamageSource("gun", weapon.getName(), this, this.getThrower()), damage);
```

Comment:
Removing the null check could cause a null pointer exception in the following cases:
- Player shoot entity, Player disconnects while the bullet is travelling
- Entity shoot player, entity get killed or despawned while the bullet is travelling
etc...